src/main/java/uk/ac/ebi/subs/validator/coordinator/ChainedValidationService.java

```diff
@@ -1,53 +1,122 @@
 
-    private Map<Class<? extends StoredSubmittable>, SubmittableRepository<? extends StoredSubmittable>> submittableRepositoryMap;
+    @NonNull
+    private List<SubmittableRepository<?>> submissionContentsRepositories;
+    @NonNull
     private SubmittableHandler submittableHandler;
 
-    public ChainedValidationService(
-            Map<Class<? extends StoredSubmittable>, SubmittableRepository<? extends StoredSubmittable>> submittableRepositoryMap,
-            SubmittableHandler submittableHandler
-    ) {
-        this.submittableRepositoryMap = submittableRepositoryMap;
-        this.submittableHandler = submittableHandler;
+    public void triggerChainedValidation(String submissionId){
+        submissionContentsRepositories.stream()
+                .flatMap(repo -> repo.streamBySubmissionId(submissionId))
+                .forEach(storedSubmittable -> revalidate(storedSubmittable,submissionId));
+    }
+
+
+    private void revalidate(StoredSubmittable storedSubmittable, String submissionId){
+        submittableHandler.handleSubmittable(
+                storedSubmittable,
+                submissionId,
+                (storedSubmittable.getDataType() == null) ? null : storedSubmittable.getDataType().getId(),
+                (storedSubmittable.getChecklist() == null) ? null : storedSubmittable.getChecklist().getId()
+        );
     }
 
     public void triggerChainedValidation(Submittable triggerSubmittable, String submissionId) {
-        streamSubmittablesInSubmissionExceptTriggerSubmittable(triggerSubmittable, submissionId)
-                .forEach(storedSubmittable -> submittableHandler.handleSubmittable(
-                        storedSubmittable,
-                        submissionId,
-                        (storedSubmittable.getDataType() == null) ? null : storedSubmittable.getDataType().getId(),
-                        (storedSubmittable.getChecklist() == null) ? null : storedSubmittable.getChecklist().getId()
-                        )
-                );
+        AbstractSubsRef ref = submittableToRef(triggerSubmittable);
+
+        submissionContentsRepositories.stream()
+                .flatMap(repo -> {
+                    List<StoredSubmittable> itemsReferencingTriggerSubmittable = ((SubmittableRepositoryCustom) repo)
+                            .findBySubmissionIdAndReference(submissionId, ref);
+                    return itemsReferencingTriggerSubmittable.stream();
+                })
+                .forEach(storedSubmittable -> revalidate(storedSubmittable,submissionId));
     }
 
-    protected Stream<? extends StoredSubmittable> streamSubmittablesInSubmissionExceptTriggerSubmittable(Submittable triggerSubmittable, String submissionId) {
-        Stream<? extends StoredSubmittable> streamSubmittablesInSubmission = streamSubmittablesInSubmission(submissionId);
+    protected AbstractSubsRef submittableToRef(Submittable submittable) {
+        AbstractSubsRef ref = new SampleGroupRef();
```

Comment:
Do we need to instantiate the `ref` here with a concrete type?
Most of the type in the following lines we would replace them with another concrete type.

Reply:
I've pushed an update to correct this